killercoda/http-scaler/autoscale.md

```diff
@@ -0,0 +1,97 @@
+
+```
+kubectl wait -ndefault --timeout=5m --for=condition=Available deployment/kedify-proxy
+kubectl wait -ndefault --timeout=5m --for=jsonpath='{.spec.rules[0].http.paths[0].backend.service.name}="kedify-proxy"' ingress/blue
+kubectl get -ndefault ingress blue -o json | jq '.spec.rules[].http.paths[].backend'
+```{{exec}}
+
+The `backend`{{}} referenced in the application `Ingress`{{}} is no longer `"blue"`{{}} but `"kedify-proxy"`{{}} instead. This means the network traffic is routed there first and then back to the application.
+
+> The `kedify-proxy`{{}} forms an [envoy proxy](https://www.envoyproxy.io/) fleet deployed/undeployed and configured dynamically by Kedify. It also forwards live metrics to KEDA for scaling. When using scale to 0, there is also one instance of centrally deployed reverse proxy called `interceptor`{{}}. This `interceptor`{{}} is a component from [HTTP Add-on](https://github.com/kedacore/http-add-on) used for caching requests during application cold starts.
+
+You can observe the scaling metrics directly from KEDA using `kubectl`{{}}
+```bash
+kubectl get --raw '/api/v1/namespaces/keda/services/keda-add-ons-http-interceptor-admin:9090/proxy/queue' | jq '.'
+```{{exec}}
+
+Right now you should see that our configured metric `RPS`{{}} is `0`{{}}, that is because we made no requests to the application since it has been configured for autoscaling.
+```json
+{
+  "default/demo": {
+    "Concurrency": 0,
+    "RPS": 0
+  }
+}
+```{{}}
+
+Let's run a few requests with `curl`{{}} and see if the metric changes
+```bash
+for i in $(seq 1 20); do curl http://blue.com; done
+```{{exec}}
+```bash
+kubectl get --raw '/api/v1/namespaces/keda/services/keda-add-ons-http-interceptor-admin:9090/proxy/queue' | jq '.'
+```{{exec}}
+
+Before the benchmark, we will make the KEDA behaviour more rapid, both scaling up and scaling down. This is unlikely something you'd want in a production environment, but it is helpful in demos and when testing KEDA behaviour.
+```
+kubectl patch scaledobject blue -n default --type=merge -p='{"spec":{"advanced":{"horizontalPodAutoscalerConfig":{"behavior":{"scaleDown":{"stabilizationWindowSeconds": 5}}}}}}'
+kubectl patch scaledobject blue -n default --type=merge -p='{"spec":{"advanced":{"horizontalPodAutoscalerConfig":{"behavior":{"scaleUp":{"stabilizationWindowSeconds": 1}}}}}}'
+kubectl patch scaledobject blue -n default --type=merge -p='{"spec":{"cooldownPeriod": 5}}'
+```{{exec}}
+
+To observe the metric changing in real time, you can run following benchmark script running [`hey`{{}}](https://github.com/rakyll/hey) under the hood:
+```bash
+/scripts/benchmark.sh
+```{{exec}}
+
+In order to stop the benchmark, hit:
+```
+# ctrl+c
+```{{exec interrupt}}
+
+Congratulations! You have successfully configured HTTP autoscaling using Kedify, you can check out our other courses at:
+https://killercoda.com/kedify/course/killercoda
```

Comment:
We should link users to our website here

Reply:
I will add kedify.io for now, will change to kedify.io/tutorials once the page is there